src/m365/spfx/commands/package/package-teams-enable.ts

```diff
@@ -0,0 +1,199 @@
+      async (args: CommandArgs) => {
+        const fullPath: string = path.resolve(args.options.filePath);
+
+        if (!fs.existsSync(fullPath)) {
+          return `Specified sppkg file with path '${fullPath}' does not exist`;
+        }
+
+        if (!fullPath.endsWith('.sppkg')) {
+          return `Specified file is not of the valid file type. Please specify a valid sppkg file`;
+        }
+
+        if (args.options.supportedHost && args.options.supportedHost.split(',').some(splittedHost => !SpfxPackageTeamsEnableCommand.allowedSupportedHosts.includes(splittedHost))) {
+          return `The supportedHost contains an invalid value. Possible supportedHost values are: ${SpfxPackageTeamsEnableCommand.allowedSupportedHosts.join(',')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let tmpDir: string | undefined = undefined;
+
+    try {
+      if (this.verbose) {
+        logger.logToStderr('Creating temp folder to save the sppkg content in');
+      }
+
+      tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'cli-spfx'));
+
+      if (this.verbose) {
+        logger.logToStderr(`Temp folder created at path: ${tmpDir}`);
+      }
+
+      const fullPath: string = path.resolve(args.options.filePath);
+      // we need this to be able to inject mock AdmZip for testing
+      /* c8 ignore next 3 */
+      if (!this.solutionZip) {
+        this.solutionZip = new AdmZip(fullPath);
+      }
+
+      this.solutionZip.extractAllTo(tmpDir);
+      const files = fs.readdirSync(tmpDir);
+
+      let fixesApplied = false;
+
+      files.forEach(file => {
+
+        if (validation.isValidGuid(file)) {
+
+          if (this.verbose) {
+            logger.logToStderr(`Found webpart folder with id ${file}`);
+          }
+
+          const dirLocation = `${tmpDir}\\${file}`;
+          const webpartFiles = fs.readdirSync(dirLocation);
+
+          webpartFiles.forEach((wpFile: string) => {
+            if (this.verbose) {
+              logger.logToStderr(`Reading contents for file ${wpFile}`);
+            }
+
+            const fileLocation = `${dirLocation}\\${wpFile}`;
+            let fileContent = fs.readFileSync(fileLocation, { encoding: 'utf-8' });
+            const regex = new RegExp('ComponentManifest=\\"[^\\"]+\\"');
+
+            if (regex.test(fileContent)) {
+
+              const matches = fileContent.match(regex);
+              const componentManifest = matches![0];
+              const componentManifestReplaced = componentManifest.replace(/&quot;/gi, '"').replace('ComponentManifest=\"', '').slice(0, -1);
+              const parsedComponentManifest: { id: string, alias: string, supportedHosts: string[] } = JSON.parse(componentManifestReplaced);
+              const supportedHostMatches = parsedComponentManifest.supportedHosts.filter(supHost => supHost.startsWith('Teams'));
+
+              if (supportedHostMatches.length === 0) {
```

Comment:
ok so this is probably something we could discuss on the issue level 😉, but still I want to ask for small clarification and your opinion (and @waldekmastykarz also yours since you are the related issue author).

So currently if the web part will already have any kind of supported host specified then we don't do anything. I kind'a have a feeling that we could include some `overwrite` option or when `fix` is used we could always overwrite the hosts with, either default or the specified supported hosts for the command. What do you think? I hope I was clear 😉. So currently, if I already have some host specified for my SPFx package then this command is useless. But if we would allow to somehow overwrite it (either when we use `fix` or add some new option) then this command would give more functionality 💪

Reply:
Why would we need to add support for overwrite? If the purpose of the command is to enable SPFx package for deployment in Teams and the web part already is enabled for use in Teams, what would be the benefit of overwriting web part's hosts? Wouldn't the net result be exactly the same (web part is still enabled for use in Teams)?

Reply:
that is not the only purpose. Besides enabling it to teams it is also possible to specify 
```
`--supportedHost`
: Comma-separated list of type of hosts to add to the webparts if not available. Possible options are `TeamsPersonalApp`, `TeamsTab` or `TeamsMeetingApp`. Defaults to `TeamsPersonalApp`.
```

Now if some app is already enabled to teams but only as a `TeamsTab` if I would run the command with the `--fix` option and the specifying supported hosts for `TeamsTab` and `TeamsMeetingApp` there will be no result as the app is already enabled and there is no way for me to extend the `supportedHost` after that.

Reply:
I wonder if we're not mixing the different use cases here. What you're suggesting goes beyond enabling the web part for deployment in Teams, and seems like basically editing web part's manifest. I wonder if that shouldn't be another - more generic command.

Reply:
Well it's not that I am suggesting it, it's more what is already done 😉 should we take it back to the issue and recheck if this should be done in a separate command?

Reply:
Sounds like a plan. Let's do that.

Reply:
ok in the related issue, we agreed that there is a place for both: this command and a separate one which will allow to set any kind of supported host. 
The question that still remains is whether should we allow to overwrite with this command in the supported host if some are already set or not? IMO I would allow it when the `fix` option is passed. Otherwise, we do not give any possibility to change it (as of now) if it was already set by the user.

Reply:
Since we let users specify the supported hosts, we should ensure that all of them are listed in the manifest. In this command, I suggest we add them to already existing hosts rather than replacing the full property, which is something we'd support in the `set` command.

Reply:
@MathijsVerbeeck so if I understood @waldekmastykarz we should allow to extend the list when user will provide a host which is not currently defined. What do you think?